selectors/selectors.md

```diff
@@ -110,295 +110,269 @@ These components imply or expand into the following things:
 - A recommended structure for implementing a selector type, with an easy to use test suite.
 
 
-## Selector Types
+Selector types
+--------------
 
-Selectors have multiple types
+### CID Selector
 
-```
-<selector-code> <selector-string-code> <selector-type>
-0, null, selects the empty set
-1, cid, selects a specific cid
-2, path, selects an ipld path
-3, glob, selects a unix-style glob
-4, multi, combines multiple selectors
-5, css, selects in css style
-6, git, selects with git tilde notation
-7, bitfield, selects all the nodes expressed in a bitfield
-```
+Selects a single CID.
 
-**TODO**: put selectors into the grand magnificent global universal multicodec table...
 
-### Cid Selector
+### Path Selector
 
-This is a basic selector, that means exactly a single CID, and thus a single graph node. Nothing more.
+Selects a single path segment.
 
-```go
-// Binary: <selector-code><length><cid-selector-code><binary-cid>
-//     or: <cid-selector-code><binary-cid>
-// String: /sel/cid/<multibase-encoded-cid>
-type CidSelector struct {
-  Cid Cid // el cid
-}
-```
 
-### Path Selector
+### Array Selector
 
-This is a selector w/ a path. Absolute ones must start with a CID
+Selects a slice or single item of an array of items.
 
-```go
-// Binary: <selector-code><length><path-selector-code><binary-cid><binary-path>
-//     or: <path-selector-code><length><binary-path>
-// String: /sel/path/<multibase-encoded-cid>/a/b/c/d/...
-type PathSelector struct {
-  Path Path
-}
-```
 
-### Glob Selector
+### Recursive Selector
 
-This is a selector with a [unix shell glob](https://en.wikipedia.org/wiki/Glob_(programming)) style thing. Absolute ones must start with a CID. It supports the following syntax:
+Uses another Selector to select things recursively. It also takes a limit on how often this recursion should happen (the limit can be set to "infinity" to keep traversing until there are no matching paths left.
 
-- `*` - means any number of single characters, except `/` -- ie matches any link within a level (as in shell `*`)
-- `**` - means any decendant (as in shell `**`)
-- `?` - matches any single character
-- `{abc,def}` - means either `abc` or `def`
-- [abc] - matches one character in the bracket set.
-- [a-z] - matches one character from the (locale-dependent) range given in the bracket.
+The recursion may also be stopped by a given CID.
 
-```go
-// Binary: <selector-code><length><glob-selector-code><binary-cid><binary-path>
-//     or: <glob-selector-code><length><binary-path>
-// String: /sel/path/<multibase-encoded-cid>/a/*/**
-type GlobSelector struct {
-    Glob Path
-}
-```
 
-### Multi Selector
```

Comment:
I found multi selectors confusing, but if "selector queries" are supposed to be a replacement for them we need to enhance this definition. I'd really like to understand what I'd use selector queries for and how they'd work (do they union/intersect? do I get to choose?)